src/core/layout/qgslayoutaligner.cpp

```diff
@@ -97,6 +99,65 @@ void QgsLayoutAligner::distributeItems( QgsLayout *layout, const QList<QgsLayout
   if ( items.size() < 2 )
     return;
 
+  double minCoord = std::numeric_limits<double>::max();
+  double maxCoord = std::numeric_limits<double>::lowest();
+  QMap< double, QgsLayoutItem * > itemCoords;
+
+  double step;
+
+  // equispaced distribution doesn't follow the same approach of the other distribution types
+  if ( distribution == DistributeHSpace || distribution == DistributeVSpace )
+  {
+    double length = 0.0;
+
+    for ( QgsLayoutItem *item : items )
+    {
+      QRectF itemBBox = item->sceneBoundingRect();
+      double item_min, item_max;
+
+      item_min = ( distribution == DistributeHSpace ? itemBBox.left() :
+                   itemBBox.top() );
+      item_max = ( distribution == DistributeHSpace ? itemBBox.right() :
+                   itemBBox.bottom() );
+
+      minCoord = std::min( minCoord, item_min );
+      maxCoord = std::max( maxCoord, item_max );
+      length += ( item_max - item_min );
+      itemCoords.insert( item_min, item );
+    }
+    step = ( maxCoord - minCoord - length ) / ( items.size() - 1 );
```

Comment:
```suggestion
    const double step = ( maxCoord - minCoord - length ) / ( items.size() - 1 );
```

(avoids an uninitialized variable)

Reply:
@nyalldawson I just copied the already existing ``double step = ...`` of the original method, I update also it.